File: prez/code/increment.c
#include <stdio.h>
#include <stdlib.h>
#define N 256

void increment(int * u, const int n, int val) {
  for(int i = 0; i < n ; ++i) u[i] += val;
}

int main()
{
  int tab[N];
  for(int i = 0; i < N; ++i)   tab[i] = 1;
  increment(tab, N, 3);
  for (int i = 0; i < N; ++i) { if (i[tab] != 4) abort();}
  return 0;
}

File: prez/code/increment.f90
program incr
  implicit none
  integer, parameter :: N = 256
  integer :: tab(N), i

  tab = 1
  call increment(tab, 3)
  if (any(tab(:) /= 4)) stop "error"
contains
  subroutine increment(u, val)
   integer, intent(inout) :: u(:)
   integer, intent(in) :: val
   integer :: i
   do i=1, size(u, 1) ! modern => write u(:) = u(:) + val w/o loop
     u(i) = u(i) + val 
   end do
  end subroutine increment
end program incr
